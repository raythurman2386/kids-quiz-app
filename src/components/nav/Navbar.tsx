import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Wrapper } from '../../styles/Nav.style';

const Navbar = ({ isLoggedIn }: any) => {
  return (
    <Wrapper data-testid='navbar-component'>
      <h1 data-testid='nav-title'>Kidspace</h1>
      <nav data-testid='nav-links'>
        <Link to='/'>Home</Link>
        {isLoggedIn ? (
          <>
            <Link to='/dashboard'>Dashboard</Link>
            <Link to='/profile'>Profile</Link>
          </>
        ) : (
          <>
            <Link to='/login'>Login</Link>
            <Link to='/register'>Register</Link>
          </>
        )}
      </nav>
    </Wrapper>
  );
};

const mapStateToProps = (state: any) => ({
  isLoggedIn: state.user.isLoggedIn,
});

export default connect(mapStateToProps, null)(Navbar);
