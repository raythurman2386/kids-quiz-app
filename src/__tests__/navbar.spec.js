import '@testing-library/jest-dom/extend-expect';

import React from 'react';
import { render } from '@testing-library/react';

import Navbar from '../components/nav/Navbar';

describe('Nav component tests', () => {
  test('Nav component displays correctly.', () => {
    const navComponent = render(<Navbar />);
    expect(navComponent.getByTestId('navbar-component')).toBeDefined();
    expect(navComponent.getByTestId('nav-title')).toHaveTextContent('Kidspace');
    expect(navComponent).toMatchSnapshot();
  });
});
