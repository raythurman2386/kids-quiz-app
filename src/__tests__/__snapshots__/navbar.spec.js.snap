// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component tests It displays correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdnylx iHbepJ"
        data-testid="navbar-component"
      >
        <h1
          data-testid="nav-title"
        >
          Kidspace
        </h1>
        <nav
          data-testid="nav-links"
        >
          <a
            href="#"
          >
            Home
          </a>
          <a
            href="#"
          >
            Dashboard
          </a>
          <a
            href="#"
          >
            Profile
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdnylx iHbepJ"
      data-testid="navbar-component"
    >
      <h1
        data-testid="nav-title"
      >
        Kidspace
      </h1>
      <nav
        data-testid="nav-links"
      >
        <a
          href="#"
        >
          Home
        </a>
        <a
          href="#"
        >
          Dashboard
        </a>
        <a
          href="#"
        >
          Profile
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Nav component tests Nav component displays correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdnylx iHbepJ"
        data-testid="navbar-component"
      >
        <h1
          data-testid="nav-title"
        >
          Kidspace
        </h1>
        <nav
          data-testid="nav-links"
        >
          <a
            href="#"
          >
            Home
          </a>
          <a
            href="#"
          >
            Dashboard
          </a>
          <a
            href="#"
          >
            Profile
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdnylx iHbepJ"
      data-testid="navbar-component"
    >
      <h1
        data-testid="nav-title"
      >
        Kidspace
      </h1>
      <nav
        data-testid="nav-links"
      >
        <a
          href="#"
        >
          Home
        </a>
        <a
          href="#"
        >
          Dashboard
        </a>
        <a
          href="#"
        >
          Profile
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
